source/Calamari.Azure/Scripts/AzureContext.ps1

```diff
@@ -118,8 +119,21 @@ Execute-WithRetry{
                     # authenticate with the Azure CLI
                     Write-Host "##octopus[stdout-verbose]"
 
-                    $env:AZURE_CONFIG_DIR = [System.IO.Path]::Combine($env:OctopusCalamariWorkingDirectory, "azure-cli")
-                    EnsureDirectoryExists($env:AZURE_CONFIG_DIR)
+                    # Config directory is set to make sure that our security is right for the step running it  
+                    # and not using the one in the default config dir to avoid issues with user defined ones 
+                    $env:AZURE_CONFIG_DIR = [System.IO.Path]::Combine($env:OctopusCalamariWorkingDirectory, "azure-cli") 
+                    EnsureDirectoryExists($env:AZURE_CONFIG_DIR) 
+ 
+                    # Set the azure extensions directory to $HOME\.azure\cliextension as that's the default, but it's getting 
+                    # overridden above when we set the azure config dir. By defining $env:AZURE_EXTENSION_DIR we can override the 
+                    # directory again back to the default 

```

Comment:
```suggestion
                    # The azure extensions directory is getting overridden above when we set the azure config dir (undocumented behavior). 
                    # Set the azure extensions directory to the value of $OctopusAzureExtensionsDirectory if specified, 
                    # otherwise, back to the default value of $HOME\.azure\cliextension.
```